tests/copyoffload/test_copyoffload_migration.py

```
@@ -3472,6 +3478,203 @@ def test_check_vms(
+    "populator_inflight_forkliftcontroller",
+    "copyoffload_ssh_key",
+    "cleanup_migrated_vms",
+)
+class TestCopyoffloadPopulatorThrottlingMigration:
+    """Copy-offload migration (MTV-696): per-ESXi-host populator throttling and sourceHost labels.
+
+    Covers MTV-5654 Test 1 + Test 3:
+    - Set controller_max_populator_inflight to a low value (2) on ForkliftController
+    - Migrate a VM with 4+ disks from a single ESXi host
+    - Verify peak populator concurrency per host respects the limit and reaches min(limit, disk_count)
+    - Verify PopulatorThrottled events on at least (disk_count − limit) PVCs and sourceHost labels on populate pods
```

Ruff (0.15.15)

[warning] Docstring contains ambiguous `−` (MINUS SIGN). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)